projects/packages/search/src/class-plan.php

```diff
@@ -16,7 +16,27 @@
  * Registers the REST routes for Search.
  */
 class Plan {
-	const JETPACK_SEARCH_PLAN_INFO_OPTION_KEY = 'jetpack_search_plan_info';
+	const JETPACK_SEARCH_PLAN_INFO_OPTION_KEY  = 'jetpack_search_plan_info';
+	const JETPACK_SEARCH_EVER_SUPPORTED_SEARCH = 'jetpack_search_ever_supported_search';
+
+	/**
+	 * Whether we have hooked the actions.
+	 *
+	 * @var boolean
+	 */
+	protected static $hooks_inited = false;
+
+	/**
+	 * Init hooks for updating plan info
+	 */
+	public function init_hooks() {
+		// Update plan info from WPCOM on Jetpack heartbeat.
+		// TODO: implement heartbeart for search.
```

Comment:
Is this comment correct? I thought this PR implemented the heartbeat?

If this PR doesn't implement the heartbeat, what does it do exactly?

Reply:
Ah, got it. Let me rephrase what you've said to see if I understood you correctly.

This PR doesn't implement heartbeat for Search because we're not emitting any data back to WPCOM servers. However, it does add periodic polling of WPCOM plan info.

Is this correct?

Reply:
>This PR doesn't implement heartbeat for Search because we're not emitting any data back to WPCOM servers. However, it does add periodic polling of WPCOM plan info.

hmmm, no. In other words, Jetpack implemented its heartbeat ( it applies 'jetpack_heartbeat' action and also provide an endpoint for WPCOM to call ). The PR hooks 'jetpack_heartbeat' to update search plan data. When we start our standalone Search plugin, there will not be such heartbeat, which is the thing we'd need to implement.

Does that make sense?

Reply:
Ah, got it! Thanks for the clarification, that makes much more sense.